src/Event/GiftcardAmountCalculateEvent.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Drupal\commerce_giftcard\Event;
+
+
+use Drupal\commerce_price\Price;
+use Symfony\Component\EventDispatcher\Event;
+
+/**
+ * Class GiftcardAmountCalculateEvent.
+ */
+class GiftcardAmountCalculateEvent extends Event {
+
+  /**
+   * The amount.
+   *
+   * @var \Drupal\commerce_price\Price|null
+   */
+  private $amount;
+
+  /**
+   * GiftcardAmountCalculateEvent constructor.
+   */
+  public function __construct($amount = NULL) {
+    $this->amount = $amount;
+  }
+
+  /**
+   * Get the amount.
+   */
+  public function getAmount() {
+    return $this->amount;
+  }
+
+  /**
+   * Amount setter.
+   */
+  public function setAmount(Price $amount) {
```

Comment:
another thing, if we want to have the ability to not set an amount in an event this needs to allow NULL or we need to check for price = 0.

Reply:
Indeed. Thanks. Decided that after I made that method :)